src/server/routes/api/v1/users.ts

```diff
@@ -5,6 +5,7 @@ import type {
   FastifyReply,
   FastifyRequest,
 } from 'fastify';
+import type { Role } from '../../../models/user';
```

Comment:
As a consequence for implementing the `enum` in the User's model file, you should remove the `type` keyword here.